sdks/python/apache_beam/yaml/yaml_provider.py

```diff
@@ -139,6 +139,21 @@ def as_provider_list(name, lst):
   return [as_provider(name, x) for x in lst]
 
 
+def parse_callable_kwargs(spec: Optional[Mapping[str, Any]]):
+  def _parse_callable_kwargs_rec(args, top_level=False):
```

Comment:
Why do we distinguish between top-level vs. not?

Reply:
For the edge case where a custom provider has a single parameter named `callable`. In this design, any time a parameter is named `callable`, it must be nested under another `callable` to be evaluated as such. For example,

```
- type: MyTransform
  config:
    callable:
      callable: len
```